TestArrayGroup/src/TestDynamicData.F90

```diff
@@ -6,24 +6,21 @@ subroutine TestArrayGroup_DynamicDataF(CCTK_ARGUMENTS)
   DECLARE_CCTK_PARAMETERS
   DECLARE_CCTK_ARGUMENTS
 
+  integer, dimension(3) :: dim3
+
+  ! RANK() is not supported by all compilers, so we fail to complile here instead
   ! Validate grid array dynamic data
-  if(RANK(test1) /= 3) then ! note rank 3 b/c of vector of rank=2 arrays
-      call CCTK_ERROR("incorrect dimension in test1 array dynamic data")
-  endif
+  dim3 = SHAPE(test1)
```

Comment:
`dim3` seems unused. Did you want to test the length of `dim3` as replacement of the rank test?

Reply:
the compiler aborts if the rank of dim3 is not correct. So while it is unused the assignment still test that `test1` is of the expected shape. I'll add a comment